cdc/log.cc

```diff
@@ -835,8 +835,14 @@ class transformer final {
         auto&& cc = _schema->clustering_key_columns();
 
         std::vector<query::clustering_range> bounds;
-        if (cc.empty()) {
+        uint32_t row_limit = query::max_rows;
+
+        const bool has_only_static_row = !p.static_row().empty() && p.clustered_rows().empty();
+        if (cc.empty() || has_only_static_row) {
             bounds.push_back(query::clustering_range::make_open_ended_both_sides());
+            if (has_only_static_row) {
+                row_limit = 1;
+            }
```

Comment:
This doesn't fix static row preimage entirely:

```
cqlsh> create table ks.t (pk int, ck int, vs int static, vc int, primary key (pk, ck)) with cdc = {'enabled':true, 'preimage': true};
cqlsh> update ks.t set vs = 0, vc = 0 where pk = 0 and ck = 0;
cqlsh> update ks.t set vs = 1, vc = 1 where pk = 0 and ck = 1;
cqlsh> select "cdc$time", "cdc$batch_seq_no", "cdc$operation", ck, vs, vc from ks.t_scylla_cdc_log ;

 cdc$time                             | cdc$batch_seq_no | cdc$operation | ck   | vs   | vc
--------------------------------------+------------------+---------------+------+------+------
 4a8cab96-5f91-11ea-1856-0dd78f1f3386 |                0 |             1 | null |    0 | null
 4a8cab96-5f91-11ea-1856-0dd78f1f3386 |                1 |             1 |    0 | null |    0
 4d78f1ca-5f91-11ea-2181-3c03a73998dc |                0 |             1 | null |    1 | null
 4d78f1ca-5f91-11ea-2181-3c03a73998dc |                1 |             1 |    1 | null |    1

(4 rows)
```

The situation happens when in the same statement you update:
1. the static row,
2. and a clustered row which doesn't exist yet, so it doesn't have preimage

Then you'll query the clustered row in pre_image_select, and the query will return no rows. So you don't get the static column's value.

Unfortunately, there is no quick way to fix this efficiently with the current state of things. In the most general case we would have to perform two pre_image selects.

We could try modifying our internal APIs so that you can submit a query which always returns the static row, even if there are no clustering keys matching your selected range.

Reply:
@piodul could you please take care of this?

Reply:
Oops, sorry, my bad. Fortunately, some time ago Vladimir Davydov already added a very convenient flag to partition slice options, so the fix was trivial :)

PR: https://github.com/scylladb/scylla/pull/5975